test/hooks.test.js

```diff
@@ -3404,3 +3408,210 @@ test('registering invalid hooks should throw an error', async t => {
+
+    sleep(500).then(() => socket.destroy()).catch(error => t.error(error))
+  })
+})
+
+test('onRequestAbort should handle errors / 1', t => {
+  const fastify = Fastify()
+
+  t.plan(2)
+  t.teardown(() => fastify.close())
+
+  fastify.addHook('onRequestAbort', function (req, done) {
+    process.nextTick(() => t.pass())
+    done(new Error('KABOOM!'))
+  })
+
+  fastify.route({
+    method: 'GET',
+    path: '/',
+    async handler (request, reply) {
+      await sleep(1000)
+      return { hello: 'world' }
+    }
+  })
+
+  fastify.listen({ port: 0 }, err => {
+    t.error(err)
+
+    const socket = connect(fastify.server.address().port)
+
+    socket.write('GET / HTTP/1.1\r\nHost: example.com\r\n\r\n')
+
+    sleep(500).then(() => socket.destroy()).catch(error => t.error(error))
+  })
+})
+
+test('onRequestAbort should handle errors / 2', t => {
+  const fastify = Fastify()
+
+  t.plan(2)
+  t.teardown(() => fastify.close())
+
+  fastify.addHook('onRequestAbort', function (req, done) {
+    process.nextTick(() => t.pass())
+    throw new Error('KABOOM!')
+  })
+
+  fastify.route({
+    method: 'GET',
+    path: '/',
+    async handler (request, reply) {
+      await sleep(1000)
+      return { hello: 'world' }
+    }
+  })
+
+  fastify.listen({ port: 0 }, err => {
+    t.error(err)
+
+    const socket = connect(fastify.server.address().port)
+
+    socket.write('GET / HTTP/1.1\r\nHost: example.com\r\n\r\n')
+
+    sleep(500).then(() => socket.destroy()).catch(error => t.error(error))
+  })
+})
+
+test('onRequestAbort should handle async errors / 1', t => {
+  const fastify = Fastify()
+
+  t.plan(2)
+  t.teardown(() => fastify.close())
+
+  fastify.addHook('onRequestAbort', async function (req) {
+    process.nextTick(() => t.pass())
+    throw new Error('KABOOM!')
+  })
+
+  fastify.route({
+    method: 'GET',
+    path: '/',
+    async handler (request, reply) {
+      await sleep(1000)
+      return { hello: 'world' }
+    }
+  })
+
+  fastify.listen({ port: 0 }, err => {
+    t.error(err)
+
+    const socket = connect(fastify.server.address().port)
+
+    socket.write('GET / HTTP/1.1\r\nHost: example.com\r\n\r\n')
+
+    sleep(500).then(() => socket.destroy()).catch(error => t.error(error))
+  })
+})
+
+test('onRequestAbort should handle async errors / 2', t => {
+  const fastify = Fastify()
+
+  t.plan(2)
+  t.teardown(() => fastify.close())
+
+  fastify.addHook('onRequestAbort', async function (req) {
+    process.nextTick(() => t.pass())
+    return Promise.reject()
+  })
+
+  fastify.route({
+    method: 'GET',
+    path: '/',
+    async handler (request, reply) {
+      await sleep(1000)
+      return { hello: 'world' }
+    }
+  })
+
+  fastify.listen({ port: 0 }, err => {
+    t.error(err)
+
+    const socket = connect(fastify.server.address().port)
+
+    socket.write('GET / HTTP/1.1\r\nHost: example.com\r\n\r\n')
+
+    sleep(500).then(() => socket.destroy()).catch(error => t.error(error))
```

Comment:
this is confusing because we have `plan(2)` and it is never executed and, if it does, the test would fail

```suggestion
    sleep(500).then(() => socket.destroy())
```